util/utilities.q

```diff
@@ -0,0 +1,159 @@
+// util/utilities.q - Utilities library
+// Copyright (c) 2021 Kx Systems Inc
+//
+// Includes range, arange, combs, eye, iMax, iMin,
+// linearSpace, shape, trainTestSplit, tab2df,
+// df2tabTimezone, df2tab
+
+\d .ml
+
+// @kind function
+// @category utilities
+// @desc Range of values
+// @param array {number[]} A numerical array 
+// @returns {float} Range of its values
+range:{[array]
+  max[array]-min array
+  }
+
+// @kind function
+// @category utilities
+// @desc Evenly-spaced values
+// @param start {number} Start of the interval (inclusive)
+// @param end {number} End of the interval (non-inclusive)
+// @param step {number} Spacing between values 
+// @return {number[]} A vector of evenly-spaced values between start and end
+//   in steps of length `step`
+arange:{[start;end;step]
+  start+step*til 0|ceiling(end-start)%step
+  }
+
+// @kind function
+// @category utilities
+// @desc Unique combinations of a vector or matrix
+// @param n {int} Number of values required for combinations
+// @param degree {int} Degree of the combinations to be produced
+// @return {int[]} Unique combinations of values from the data 
+combs:{[n;degree]
+  flip(degree-1)i.combFunc[n]/enlist til n
+  }
+
+// @kind function
+// @category utilities
+// @desc Create identity matrix 
+// @param n {int} Width/height of identity matrix
+// @return {int[]} Identity matrix of height/width n
+eye:{[n]
+  @[n#0.;;:;1.]each til n
+  }
+
+// @kind function
+// @category utilities
+// @desc Index of maximum element of a list
```

Comment:
Probably appropriate to say that this is the `index of the first occurrence of the maximum value of a list`, it's important to know that it won't get you all of the values

The same holds for `iMin` below